yt/frontends/stream/data_structures.py

```diff
@@ -291,6 +294,25 @@ def __init__(
             default_species_fields=default_species_fields,
         )
 
+    def _check_for_mks_em_units(self):
+        from yt.units.dimensions import current_mks
+
+        # We assume CGS EM units by default, unless
+        # magnetic_unit is explicitly defined within MKSA
+        magnetic_unit = self.stream_handler.code_units[-1]
```

Comment:
I'm not too optimistic here but is there any way to make this less fragile ? If we ever append another unit to `self.stream_handler.code_units`, or change their order, this would break, possibly silently.

Reply:
@neutrinoceros it's not a bad idea (I thought this myself), but that's a much bigger job than this PR is trying to solve, as it would touch quite a bit of code in the stream frontend.

Reply:
okay so I've looked into this some more and found that the code breaks (badly) in every scenario I could think of
- appending a new unit
- changing their order
- dropping the magnetic unit

So it's still not ideal, but at least I'm now confident that this will not break silently if we ever change something around it.

Reply:
We should definitely change it--I think it deserves a new PR on its own.